[videoUtils] Add `ConvertSeqToVideo` and `ConvertVideoToSeq` nodes

cbentejac opened the pull request:
This PR adds two nodes, "ConvertSeqToVideo" and "ConvertVideoToSeq", that handle the direct conversion from sequences to videos and from videos to sequences, without any compression or resizing options.

cbentejac committed:
[videoUtils] Add `ConvertSeqToVideo` and `ConvertVideoToSeq` nodes

fabiencastan commented on meshroom/videoUtils/ConvertVideoToSeq.py:
```diff
@@ -0,0 +1,57 @@
+            name="outputFolder",
+            label="Output Folder",
+            description="Folder where to save the frames.",
+            value=lambda attr: os.path.join("{nodeCacheFolder}", "frames"),
```
```suggestion
            value="{nodeCacheFolder}",
```

fabiencastan commented on meshroom/videoUtils/ConvertVideoToSeq.py:
```diff
@@ -0,0 +1,57 @@
+        if not os.path.exists(folder):
+            os.makedirs(folder)
+        desc.CommandLineNode.processChunk(self, chunk)
+
```
```suggestion
```

fabiencastan replied:
we can avoid the mkdir if we do not create a useless subfolder.

fabiencastan replied:
Maybe keep the subfolder for now to avoid compatibility issues.

fabiencastan commented on meshroom/videoUtils/ConvertVideoToSeq.py:
```diff
@@ -0,0 +1,57 @@
+
+
+class ConvertVideoToSeq(desc.CommandLineNode):
+    commandLine = "ffmpeg -y -i {videoFileValue}"
```
```suggestion
    commandLine = "ffmpeg -y -i {videoFileValue} {outputFolderValue}/{patternValue}"
```

fabiencastan commented on meshroom/videoUtils/ConvertVideoToSeq.py:
```diff
@@ -0,0 +1,57 @@
+        filePattern = chunk.node.attribute("pattern").value
+
+        suffix = f" {os.path.join(outputFolder,filePattern)}"
+        return desc.CommandLineNode.buildCommandLine(self, chunk) + suffix
```
```suggestion
```

fabiencastan commented on meshroom/videoUtils/ConvertSeqToVideo.py:
```diff
@@ -0,0 +1,40 @@
+            name="outputVideo",
+            label="Output Video",
+            description="Output video file.",
+            value=os.path.join("{nodeCacheFolder}", "video.mp4"),
```
```suggestion
            value="{nodeCacheFolder}/video.mp4"),
```

cbentejac replied:
Done

fabiencastan commented on meshroom/videoUtils/ConvertVideoToSeq.py:
```diff
@@ -0,0 +1,57 @@
+            label="Output Folder",
+            description="Folder where to save the frames.",
+            value=lambda attr: os.path.join("{nodeCacheFolder}", "frames"),
+        ),
```
It would be good to add:
```suggestion
        ),
        desc.File(
            name="outputSeq",
            label="Image Sequence",
            semantic="sequence",
            description="Output image sequence.",
            value="{nodeCacheFolder}/{patternValue}",
            group="",
        ),
```

cbentejac committed:
[videosUtils] Simplify `ConvertSeqToVideo` and `ConvertVideoToSeq`

Additionally, add an output parameter to `ConvertVideoToSeq` to
visualise the generated sequence.

cbentejac committed:
[README] Update package name
